aws/resource_aws_api_gateway_rest_api.go

```diff
@@ -166,6 +166,35 @@ func resourceAwsApiGatewayRestApiUpdateOperations(d *schema.ResourceData) []*api
 		})
 	}
 
+	if d.HasChange("binary_media_types") {
+		o, n := d.GetChange("binary_media_types")
+		prefix := "binaryMediaTypes"
+
+		old := o.([]interface{})
+		new := n.([]interface{})
+
+		// Remove every binary media types. Simpler to remove and add new ones,
+		// since there are no replacings.
+		for _, v := range old {
+			m := v.(string)
+			operations = append(operations, &apigateway.PatchOperation{
+				Op:   aws.String("remove"),
+				Path: aws.String(fmt.Sprintf("/%s/%s", prefix, strings.Replace(m, "/", "~1", -1))),
+			})
+		}
+
+		// Handle additions
+		if len(new) > 0 {
+			for _, v := range new {
+				m := v.(string)
+				operations = append(operations, &apigateway.PatchOperation{
+					Op:   aws.String("add"),
+					Path: aws.String(fmt.Sprintf("/%s/%s", prefix, strings.Replace(m, "/", "~1", -1))),
+				})
+			}
+		}
+	}
```

Comment:
Computing the difference, instead of removing all old ones & adding all new ones would be a bit more efficient, but the user probably won't ever have too long list of types, so it probably doesn't matter too much - so feel free to ignore me.

_thinking out laud_ 🤔